apps/dashboard/src/components/workflow-editor/steps/components/preview-step-results-section.tsx

```diff
@@ -0,0 +1,144 @@
+
+  const toggleStepOpen = useCallback((stepId: string) => {
+    setOpenSteps((prev) => ({ ...prev, [stepId]: !prev[stepId] }));
+  }, []);
+
+  const handleStepDataChange = useCallback(
+    (stepId: string, updatedStepData: any) => {
+      const stepType = getStepType(workflow, stepId);
+
+      let finalStepData = updatedStepData;
+
+      // Apply digest synchronization if it's a digest step
+      if (stepType === 'digest') {
+        const currentStepData = localParsedData.steps?.[stepId] || {};
+        finalStepData = synchronizeDigestStepData(updatedStepData, currentStepData, workflow?.payloadExample);
+      }
+
+      const updatedSteps = {
+        ...(localParsedData.steps || {}),
+        [stepId]: finalStepData,
+      };
+
+      onUpdate('steps', updatedSteps);
+    },
+    [workflow, localParsedData.steps, onUpdate]
+  );
+
+  const getCurrentStepData = useCallback(
+    (stepId: string, stepData: any) => {
+      const stepType = getStepType(workflow, stepId);
+
+      // For digest steps, ensure the data is always synchronized
+      if (stepType === 'digest' && stepData && 'eventCount' in stepData && 'events' in stepData) {
+        return synchronizeDigestStepData(stepData, {}, workflow?.payloadExample);
+      }
+
+      return stepData;
+    },
+    [workflow]
+  );
+
+  const stepEntries = Object.entries(localParsedData.steps || {});
+
+  return (
+    <AccordionItem value="step-results" className={ACCORDION_STYLES.itemLast}>
+      <AccordionTrigger className={ACCORDION_STYLES.trigger}>
+        <div className="flex items-center gap-0.5">
+          Step results
+          <Tooltip>
+            <TooltipTrigger asChild>
+              <span className="text-foreground-400 inline-block hover:cursor-help">
+                <RiInformationLine className="size-3" />
+              </span>
+            </TooltipTrigger>
+            <TooltipContent className="max-w-xs">
+              Output data from previous steps in the workflow that can be used in subsequent steps.
+            </TooltipContent>
+          </Tooltip>
+        </div>
+      </AccordionTrigger>
+      <AccordionContent className="flex flex-col gap-2">
+        {stepEntries.length > 0 ? (
+          <div className="w-full">
+            {stepEntries.map(([stepId, stepData]) => {
+              const stepType = getStepType(workflow, stepId);
+              const StepIcon = getStepTypeIcon(stepType);
+              const stepName = getStepName(workflow, stepId);
+              const isCurrentStep = stepId === currentStepId;
+              const isOpen = openSteps[stepId] || false;
+              const currentStepData = getCurrentStepData(stepId, stepData);
+
+              return (
+                <div key={stepId}>
+                  <button
+                    type="button"
+                    onClick={() => toggleStepOpen(stepId)}
+                    className="flex w-full items-center gap-2 py-1.5 transition-colors hover:bg-neutral-50"
```

Comment:
use `interpolate-size` new css thing to animate this open/close states, check [this comment](https://github.com/novuhq/novu/pull/8310/files#r2090722497)